FnCAS/fncas/optimize.h

```diff
@@ -309,6 +374,7 @@ template <OptimizationDirection DIRECTION>
 struct OptimizeImpl<GradientDescentOptimizerSelector, DIRECTION> {
   template <typename ORIGINAL_F, typename F, typename G>
   static OptimizationResult RunOptimize(const Optimizer<ORIGINAL_F, DIRECTION>& super,
+                                        const ORIGINAL_F& original_f,
```

Comment:
Why do you need `original_f` here? Just to use in `TrackIteration`? What's the difference between "original" and "non-original" f? Do they result in different values for the same point?